### PR TITLE
chore: bump golangci-lint and fix govet issues

### PR DESCRIPTION
**Pull request by milosgajdos**
The latest `golangci-lint` spits out some `govet` issues. This PR fixes them. We are also bumping the linter version.

### CODE REVIEW

File: registry/handlers/api_test.go [hunk 1]
```diff
@@ -1107,7 +1107,7 @@ func testBlobDelete(t *testing.T, env *testEnv, args blobArgs) {
 	ref, _ := reference.WithDigest(imageName, layerDigest)
 	layerURL, err := env.builder.BuildBlobURL(ref)
 	if err != nil {
-		t.Fatalf(err.Error())
+		t.Fatal(err.Error())
```
Comment (thaJeztah) [inline registry/handlers/api_test.go:1110]: As we're updating these, these can be simplified to;

```go
t.Fatal(err)
```

(same for the other ones)

Comment (milosgajdos) [reply registry/handlers/api_test.go:1110]: Actually, that's a great shout. I should have realised that. Thanks, will update

Comment (thaJeztah) [reply registry/handlers/api_test.go]: Yeah, it's one of those cases that either way "works" so may not be worth the code-churn to update existing uses, but as we're touching these lines, might as well do it ☺️

**Comment by milosgajdos** [comment]
I went on a massive rampage and fixed all the `testing.T` formatting errors. Should be fine now! 🤞

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by joaodrp — APPROVED** [review]
LGTM

**Pull request merged**